### ISSUE TITLE
SectionIV.B "here F takes the controllable canonical form." "H to take the inverse canonical form"

### ISSUE THREAD

**Issue by Gaming722** [KalmanNet/KalmanNet_TSP#30]
https://github.com/KalmanNet/KalmanNet_TSP/blob/cc097e125affbf9d5bc38f08cefc27ce86f34762/Extended_data.py#L22-L56

Dear authors,
Thank you for your excellent work, I have some questions as follows:
 1.  In your paper Section IV.B, it is mentioned that "here F takes the controllable canonical form." and "we set H to take the inverse canonical form". But in above codes, the F does not appear to be a controllable canonical form, and I also want to know what "inverse canonical form" means.
 2. In Section IV.B 2), "Neural Model Selection", you compared KalmanNet with Vanilla RNN and MB RNN, I couldn't find related codes, I want to know if Vanilla RNN and MB RNN have same setting C1 as KalmanNet, i.e., architecture1 with input features {F2, F4} and with training algorithm V3.

Looking forward to your reply,
Thank you.

**Comment by XiaoyongNI** [comment]
Dear Jiaming,

Thanks for your interest in our work!

Regarding your questions,
1. I think it is a typo here, since our model does not consider control inputs. Sorry for the misunderstanding. By "inverse canonical form" we mean the first row and the sub-diagonal are all ones while the others are zeros.
2. You can find the code here: [Vanilla_rnn.py](https://github.com/KalmanNet/KalmanNet_TSP/blob/architecture-%232/Vanilla_rnn.py). Both Vanilla RNN and MB RNN use architecture1, but they do not use features in the form of differences like F1-4, they directly use previous posterior \hat{x}_{t-1} and current observation y_t. (this choice of feature is mainly because they are not learning a "compensation" like Kalman Gain anymore, they directly learn the output posterior \hat{x} instead). The training algorithm depends on the complexity of the dataset. For example, if we are training on a simple linear model with relatively short trajectory, we will use V1.

Best regards,
Xiaoyong

**Comment by Gaming722** [comment]
Dear Xiaoyong,

Thanks for your patient and informative responses to my questions!

I have an additional question about whether the term 'inverse canonical form' is a concept in control theory. I couldn't find its definition. I understand that my question may seem basic, and I appreciate your patience and understanding.

Best regards,
Jiaming

**Comment by XiaoyongNI** [comment]
Dear Jiaming,

From my knowledge, this is not a formal term in control theory.

**Comment by jiangyy630** [comment]
> Dear Jiaming,
> 
> Thanks for your interest in our work!
> 
> Regarding your questions,
> 
> 1. I think it is a typo here, since our model does not consider control inputs. Sorry for the misunderstanding. By "inverse canonical form" we mean the first row and the sub-diagonal are all ones while the others are zeros.
> 2. You can find the code here: [Vanilla_rnn.py](https://github.com/KalmanNet/KalmanNet_TSP/blob/architecture-%232/Vanilla_rnn.py). Both Vanilla RNN and MB RNN use architecture1, but they do not use features in the form of differences like F1-4, they directly use previous posterior \hat{x}_{t-1} and current observation y_t. (this choice of feature is mainly because they are not learning a "compensation" like Kalman Gain anymore, they directly learn the output posterior \hat{x} instead). The training algorithm depends on the complexity of the dataset. For example, if we are training on a simple linear model with relatively short trajectory, we will use V1
> 
> Best regards, Xiaoyong

So in the default branch 'main', is the canonical form F0 mentioned in the corresponding section the identity matrix?

**Comment by jiangyy630** [comment]
Dear author, I have a question that I don't understand.In the default branch 'main', is the canonical form F0 mentioned in the corresponding section the identity matrix?

**Comment by XiaoyongNI** [comment]
No, it is the canonical F0, see https://github.com/KalmanNet/KalmanNet_TSP/blob/main/Simulations/Linear_canonical/parameters.py#L28
The first row is all ones.

Best,
Xiaoyong
